recipes/spdlog/all/conanfile.py

```diff
@@ -63,24 +61,19 @@ def requirements(self):
             self.requires("fmt/6.0.0", transitive_headers=True)
 
     def package_id(self):
-        if self.options.header_only:
+        if self.info.options.header_only:
```

Comment:
what does it change? what is the difference for conan?

Reply:
Conan 2.0 disallows access to `self.options` from `package_id()`:

```
ERROR: spdlog/1.11.0: Error in package_id() method, line 68
        if self.options.header_only:
        ConanException: 'self.options' access in 'package_id()' method is forbidden
```

Reply:
Why?

Reply:
One word reply’s are generally seen as rude so I love to encourage everyone to use short bullet point style sentences, make its much more agreeable to answer helpfully :)

In 2.0 we will be getting a `compatibility.py` plugin which will using the `self.info` object to compare against the potential other settings to find a match.

This is different from the setting exact settings the graph is workign with which is the `self.settings` and `self.options` we are use to.

Only the package ID is used for the compatibility AFAIK (this is new too me as well) so that's were need to make the adjustments for the "external comparison" - thats how I understand it so far :)

Reply:
If I read https://github.com/conan-io/conan/issues/12693 correctly, it seems that using self.info in conan-center recipes for external comparison in package_id() is premature.

Reply:
There's still an open question around this and I see people asking for more recipes to test with the current beta6 so I would love to get this out.

Theres lot of other improvements to go in as well